account_commission_sales_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from odoo.tests import Form
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    def _prepare_invoice(self, journal, product, date=False):
```

Comment:
Why doing this complete override instead of doing:

```python
    def _prepare_invoice(self, journal, product, date=False):
        res = super()._prepare_invoice(journal, product, date)
        (add our logic here, updating the content of res)
        return res
```

Reply:
I tried to update the content of res like below but that doesn't work. That why I decided to override completely.
   def _prepare_sales_fee_product(self):
        sales_fee_product = self.env["product.product"].search(
            [("sales_fee", "=", True)], limit=1
        )
        if not sales_fee_product:
            raise UserError(
                _(
                    "You don't have the sales fee product. Please create sales fee product."
                )
            )
        return {
            "product_id": sales_fee_product.id,
            "quantity":len(self.line_ids),
            "price_unit": sales_fee_product.standard_price * -1,
            "settlement_id": self.id,
        }
       
    def _prepare_invoice(self, journal, product, date=False):
        res = super()._prepare_invoice(journal, product, date)
        if res["invoice_line_ids"]:
            res["invoice_line_ids"].append((0, 0, self._prepare_sales_fee_product()))
        return res

Reply:
@AungKoKoLin1997 
```suggestion
    def _prepare_invoice(self, journal, product, date=False):
        res = super()._prepare_invoice(journal, product, date)
        if res["invoice_line_ids"]:
            res["invoice_line_ids"].append((0, 0, self._prepare_sales_fee_product()))
            if "line_ids" in res:
                res.pop("line_ids", None)
        return res
```
Try this.  It seems that we need to clear `line_ids` for the `invoice_line_ids` value to survive. `invoice_line_ids` is cleared when `line_ids` is available in _move_autocomplete_invoice_lines_create() which is hooked in create() of account.move.

https://github.com/odoo/odoo/blob/a830889d12027b33cea9d81a4b47190062409779/addons/account/models/account_move.py#L2187

_values_to_save() was not the direct cause of the problem, however it does create a bit confusing situation by populating the `line_ids` value.  I'm not sure if the use of _values_to_save() in account_commission is actually appropriate but we will live with it for now.

Reply:
Or simply update `line_ids` instead of `invoice_line_ids` and see if it works nicely.